source/FreeRTOS_DHCPv6.c

```diff
@@ -448,7 +582,7 @@ static BaseType_t xDHCPv6ProcessEndPoint_HandleAdvertise( NetworkEndPoint_t * px
 
     #if ( ipconfigUSE_DHCP_HOOK != 0 ) && ( ipconfigIPv4_BACKWARD_COMPATIBLE != 1 )
         /* Ask the user if a DHCP request is required. */
-        eAnswer = xApplicationDHCPHook_Multi( eDHCPPhasePreRequest, pxEndPoint, &( pxDHCPMessage->xIPAddress ) );
+        eAnswer = xApplicationDHCPHook_Multi( eDHCPPhasePreRequest, pxEndPoint, ( IP_Address_t * ) &( pxDHCPMessage->xIPAddress ) );

```

Comment:
Is this for MISRA?

Reply:
For maintainability, changed the address type in pxDHCPMessage from IPv6_Address_t to IP_Address_t in the latest commit.